pkg/rancher-desktop/main/deploymentProfiles.ts

```diff
@@ -192,67 +161,257 @@ function parseJsonFiles(rootPath: string, defaultsPath: string, lockedPath: stri
-      if (typeof schemaVal === 'boolean') {
-        if (typeof regValue === 'number') {
-          regValue = regValue !== 0;
-        } else {
-          console.debug(`Deployment Profile expected boolean value for key ${ k }`);
-          regValue = false;
+      const innerKey = nativeReg.openKey(regKey, registryKey, nativeReg.Access.READ);
+
+      if (!innerKey) {
+        continue;
+      }
+      try {
+        regValue = this.readRegistryUsingSchema(schemaVal, innerKey, pathParts.concat([schemaKey]));
+      } finally {
+        nativeReg.closeKey(innerKey);
+      }
+      if (Object.keys(regValue).length) {
+        newObject[schemaKey] = regValue;
+      }
+    }
+    for (const { schemaKey, registryKey } of userDefinedObjectKeys) {
+      const innerKey = nativeReg.openKey(regKey, registryKey, nativeReg.Access.READ);
+
+      if (innerKey === null) {
+        console.error(`No value for registry object ${ this.fullRegistryPath(pathParts) }\\${ registryKey }`);
+        continue;
+      }
+      try {
+        regValue = this.readRegistryObject(innerKey, pathParts.concat([schemaKey]), true);
+      } catch (err: any) {
+        console.error(`Error getting registry object for ${ this.fullRegistryPath(pathParts) }\\${ registryKey }: `, err);
+      } finally {
+        nativeReg.closeKey(innerKey);
+      }
+      if (regValue) {
+        newObject[schemaKey] = regValue;
+      }
+    }
+    const unknownValueNames: string[] = [];
+    const registryValueNames = nativeReg.enumValueNames(regKey);
+    const registryValueNamesFixedCase = registryValueNames.map(k => fixProfileKeyCase(k, schemaKeys));
+
+    registryValueNamesFixedCase.forEach((key, index) => {
+      const originalName = registryValueNames[index];
+
+      if (key === null) {
+        unknownValueNames.push(originalName);
+      } else {
+        regValue = this.readRegistryValue(schemaObj[key], regKey, pathParts, originalName);
+        if (regValue !== null) {
+          (newObject as Record<string, any>)[key] = regValue;
         }
       }
+    });
+    if (unknownValueNames.length > 0) {
+      console.error(`Unrecognized value names in registry at ${ this.fullRegistryPath(pathParts) }: [${ unknownValueNames.join(', ') }]`);
     }
-    if (regValue !== null) {
-      newObject ??= {};
-      (newObject as Record<string, any>)[k] = regValue;
+
+    return newObject;
+  }
+
+  readRegistryObject(regKey: nativeReg.HKEY, pathParts: string[], isUserDefinedObject = false) {
+    const newObject: Record<string, string[] | string | boolean | number> = {};
+
+    for (const k of nativeReg.enumValueNames(regKey)) {
+      let newValue = this.readRegistryValue(undefined, regKey, pathParts, k, isUserDefinedObject);
+
+      if (newValue !== null) {
+        if (isUserDefinedObject) {
+          switch (newValue) {
+          case 0:
+            newValue = false;
+            break;
+          case 1:
+            newValue = true;
+          }
```

Comment:
Actually, any other numeric value should currently be considered an error, because we only have `string:string` and `string:boolean` user-defined objects.